tests/run.sh

```diff
@@ -153,16 +153,44 @@ if [[ -z "$BUILDDIR" ]] && [[ -n "$DOWNLOAD_REQUIREMENTS" ]]; then
+# the argument for the /extract_fs is the prefix of the image name
+# as per QA-387 now we are explicitly extracting a clean image
+# that is: an image which is crafted to be the one we update to.
+IMG_PREFIX="clean-"
 mkdir -p output
-docker run --rm --privileged --entrypoint /extract_fs -v $PWD/output:/output \
-       mendersoftware/mender-client-qemu:$(../extra/release_tool.py --version-of mender-client-qemu --version-type docker)
-docker run --rm --privileged --entrypoint /extract_fs -v $PWD/output:/output \
-        mendersoftware/mender-client-qemu-rofs:$(../extra/release_tool.py --version-of mender-client-qemu-rofs --version-type docker)
-docker run --rm --privileged --entrypoint /extract_fs -v $PWD/output:/output \
-        registry.mender.io/mendersoftware/mender-gateway-qemu-commercial:$(../extra/release_tool.py --version-of mender-gateway --version-type docker)
+docker run --rm --privileged --entrypoint /extract_fs -v "${PWD}"/output:/output \
+       mendersoftware/mender-client-qemu:$(../extra/release_tool.py --version-of mender-client-qemu --version-type docker) "${IMG_PREFIX}"
+docker run --rm --privileged --entrypoint /extract_fs -v "${PWD}"/output:/output \
+        mendersoftware/mender-client-qemu-rofs:$(../extra/release_tool.py --version-of mender-client-qemu-rofs --version-type docker) "${IMG_PREFIX}"
+docker run --rm --privileged --entrypoint /extract_fs -v "${PWD}"/output:/output \
+        registry.mender.io/mendersoftware/mender-gateway-qemu-commercial:$(../extra/release_tool.py --version-of mender-gateway --version-type docker) "${IMG_PREFIX}"
 mv output/* .
 rmdir output
 
+# check if the expected artifact_name is present
+expected_artifact_info="artifact_name=mender-image-clean"
+for i in ${IMG_PREFIX}*; do
+ mv -v "$i" "${i/${IMG_PREFIX}/}"
+ mender-artifact write rootfs-image -t beaglebone -n release-1 --software-version rootfs-v1 -f "${i/${IMG_PREFIX}/}" -o "/tmp/${i/${IMG_PREFIX}/}.mender"
+ actual_artifact_info=`mender-artifact cat "/tmp/${i/${IMG_PREFIX}/}.mender:/etc/mender/artifact_info"`
+ [[ "${expected_artifact_info}" == "${actual_artifact_info}" ]] || { echo "mismatched artifact_info. expected: $expected_artifact_info got: ${actual_artifact_info}"; exit 1; };
+ rm -f "/tmp/${i/${IMG_PREFIX}/}.mender"
+done
+
+# check if artifact_info from inside the image we run is different than the one we update to
+# the condition/naming is misleading since we expect the $expected_artifact_info to not be equal to 
+# $artifact_info in the below images
+for image_tag in mendersoftware/mender-client-qemu:$(../extra/release_tool.py --version-of mender-client-qemu --version-type docker) mendersoftware/mender-client-qemu-rofs:$(../extra/release_tool.py --version-of mender-client-qemu-rofs --version-type docker) registry.mender.io/mendersoftware/mender-gateway-qemu-commercial:$(../extra/release_tool.py --version-of mender-gateway --version-type docker); do
+ id=`docker run --rm -d "${image_tag}"`
```

Comment:
I thought `--rm` and `-d` were mutually exclusive?

Reply:
it works, in a sense: it runs and returns id, the other question is whether it removes when stopped -- I have no idea, I assume it does. this whole section is now removed, as per you suggestions. see `test_artifact_name_rofs` and `test_artifact_name` tests.